spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -212,7 +211,7 @@ public Set<TableCapability> capabilities() {
 
   @Override
   public MetadataColumn[] metadataColumns() {
-    DataType sparkPartitionType = SparkSchemaUtil.convert(Partitioning.partitionType(table()));
+    DataType sparkPartitionType = SparkSchemaUtil.convert(table().spec().partitionType());
```

Comment:
Wasn't this the unioned spec type on purpose?

Reply:
Reading it a second time, and looking at the failing test, it makes sense to me:

https://github.com/apache/iceberg/blob/dbb8a404f6632a55acb36e949f0e7b84b643cede/spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkMetadataColumns.java#L135-L166

We want to be able to jump back to earlier specs, and therefore we include them in the struct of the metadata column.